Release v2.7.0rc0

agoscinski opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
@conda-forge-admin, please rerender

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * ℹ️ `noarch: python` recipes should usually follow the syntax in our [documentation](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python) for specifying the Python version.
   - For the `host` section of `<output 0` output, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 0` output, you should usually use the pin `python >={{ python_min }}` for the `python` entry.
   - For the `test.requires` section of `<output 0` output, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 1` output, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 1` output, you should usually use the pin `python >={{ python_min }}` for the `python` entry.
   - For the `test.requires` section of `<output 1` output, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 2` output, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 2` output, you should usually use the pin `python >={{ python_min }}` for the `python` entry.
   - For the `test.requires` section of `<output 2` output, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 3` output, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 3` output, you should usually use the pin `python >={{ python_min }}` for the `python` entry.
   - For the `test.requires` section of `<output 3` output, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 4` output, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 4` output, you should usually use the pin `python >={{ python_min }}` for the `python` entry.
   - For the `test.requires` section of `<output 4` output, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - If the package requires a newer Python version than the currently supported minimum version on `conda-forge`, you can override the `python_min` variable by adding a Jinja2 `set` statement at the top of your recipe (or using an equivalent `context` variable for v1 recipes).
 * ℹ️ PyPI default URL is now pypi.org, and not pypi.io. You may want to update the default source url.
 * ℹ️ top-level output has some malformed specs:
- In section run: `python~=3.9`, `alembic~=1.2`, `archive-path~=0.4.2`, `asyncssh~=2.19.0`, `circus~=0.19.0`, `click-spinner~=0.1.8`, `click~=8.1`, `disk-objectstore~=1.3.0`, `get-annotations~=0.1`, `python-graphviz~=0.19`, `plumpy~=0.25.0`, `ipython>=7`, `jedi<0.19`, `jinja2~=3.0`, `kiwipy[rmq]~=0.8.4`, `importlib-metadata~=6.0`, `numpy~=1.21`, `paramiko~=3.0`, `pgsu~=0.3.0`, `psutil~=5.6`, `psycopg[binary]~=3.0`, `pydantic~=2.4`, `pytz~=2021.1`, `pyyaml~=6.0`, `requests~=2.0`, `sqlalchemy~=2.0`, `tabulate<0.10.0,>=0.8.0`, `tqdm~=4.45`, `typing-extensions~=4.0`, `upf_to_json~=0.9.2`, `wrapt~=1.11`, `chardet~=5.2.0`
aiida-core output has some malformed specs:
- In section run: `python~=3.9`, `alembic~=1.2`, `archive-path~=0.4.2`, `asyncssh~=2.19.0`, `circus~=0.19.0`, `click-spinner~=0.1.8`, `click~=8.1`, `disk-objectstore~=1.3.0`, `get-annotations~=0.1`, `python-graphviz~=0.19`, `plumpy~=0.25.0`, `ipython>=7`, `jedi<0.19`, `jinja2~=3.0`, `kiwipy[rmq]~=0.8.4`, `importlib-metadata~=6.0`, `numpy~=1.21`, `paramiko~=3.0`, `pgsu~=0.3.0`, `psutil~=5.6`, `psycopg[binary]~=3.0`, `pydantic~=2.4`, `pytz~=2021.1`, `pyyaml~=6.0`, `requests~=2.0`, `sqlalchemy~=2.0`, `tabulate<0.10.0,>=0.8.0`, `tqdm~=4.45`, `typing-extensions~=4.0`, `upf_to_json~=0.9.2`, `wrapt~=1.11`, `chardet~=5.2.0`
Requirement spec fields should match the syntax `name [version [build]]`to avoid known issues in conda-build. For example, instead of `name =version=build`, use `name version.* build`. There should be no spaces between version operators and versions either: `python >= 3.8` should be `python >=3.8`.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/14861945803](https://github.com/conda-forge/conda-forge-webservices/actions/runs/14861945803). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you but ran into some issues. Please check the [output logs](https://github.com/conda-forge/conda-forge-webservices/actions/runs/14861953113) of the GitHub Actions workflow below for more details. You can also ping `conda-forge/core` (using the `@` notation) for further assistance or you can try [rerendering locally](https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-with-conda-smithy-locally).


The following suggestions might help debug any issues:
* Is the `recipe/{meta.yaml,recipe.yaml}` file valid?
* If there is a `recipe/conda-build-config.yaml` file in the feedstock make sure that it is compatible with the current [global pinnnings](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml).
* Is the fork used for this PR on an organization or user GitHub account? Automated rerendering via the webservices admin bot only works for user GitHub accounts.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/14861953113](https://github.com/conda-forge/conda-forge-webservices/actions/runs/14861953113). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you but ran into some issues. Please check the [output logs](https://github.com/conda-forge/conda-forge-webservices/actions/runs/14861953099) of the GitHub Actions workflow below for more details. You can also ping `conda-forge/core` (using the `@` notation) for further assistance or you can try [rerendering locally](https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-with-conda-smithy-locally).


The following suggestions might help debug any issues:
* Is the `recipe/{meta.yaml,recipe.yaml}` file valid?
* If there is a `recipe/conda-build-config.yaml` file in the feedstock make sure that it is compatible with the current [global pinnnings](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml).
* Is the fork used for this PR on an organization or user GitHub account? Automated rerendering via the webservices admin bot only works for user GitHub accounts.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/14861953099](https://github.com/conda-forge/conda-forge-webservices/actions/runs/14861953099). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you but ran into some issues. Please check the [output logs](https://github.com/conda-forge/conda-forge-webservices/actions/runs/14861957688) of the GitHub Actions workflow below for more details. You can also ping `conda-forge/core` (using the `@` notation) for further assistance or you can try [rerendering locally](https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-with-conda-smithy-locally).


The following suggestions might help debug any issues:
* Is the `recipe/{meta.yaml,recipe.yaml}` file valid?
* If there is a `recipe/conda-build-config.yaml` file in the feedstock make sure that it is compatible with the current [global pinnnings](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml).
* Is the fork used for this PR on an organization or user GitHub account? Automated rerendering via the webservices admin bot only works for user GitHub accounts.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/14861957688](https://github.com/conda-forge/conda-forge-webservices/actions/runs/14861957688). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you but ran into some issues. Please check the [output logs](https://github.com/conda-forge/conda-forge-webservices/actions/runs/14861962257) of the GitHub Actions workflow below for more details. You can also ping `conda-forge/core` (using the `@` notation) for further assistance or you can try [rerendering locally](https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-with-conda-smithy-locally).


The following suggestions might help debug any issues:
* Is the `recipe/{meta.yaml,recipe.yaml}` file valid?
* If there is a `recipe/conda-build-config.yaml` file in the feedstock make sure that it is compatible with the current [global pinnnings](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml).
* Is the fork used for this PR on an organization or user GitHub account? Automated rerendering via the webservices admin bot only works for user GitHub accounts.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/14861962257](https://github.com/conda-forge/conda-forge-webservices/actions/runs/14861962257). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you but ran into some issues. Please check the [output logs](https://github.com/conda-forge/conda-forge-webservices/actions/runs/14861962271) of the GitHub Actions workflow below for more details. You can also ping `conda-forge/core` (using the `@` notation) for further assistance or you can try [rerendering locally](https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-with-conda-smithy-locally).


The following suggestions might help debug any issues:
* Is the `recipe/{meta.yaml,recipe.yaml}` file valid?
* If there is a `recipe/conda-build-config.yaml` file in the feedstock make sure that it is compatible with the current [global pinnnings](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml).
* Is the fork used for this PR on an organization or user GitHub account? Automated rerendering via the webservices admin bot only works for user GitHub accounts.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/14861962271](https://github.com/conda-forge/conda-forge-webservices/actions/runs/14861962271). Examine the logs at this URL for more detail.</sub>

agoscinski commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you but ran into some issues. Please check the [output logs](https://github.com/conda-forge/conda-forge-webservices/actions/runs/14862725282) of the GitHub Actions workflow below for more details. You can also ping `conda-forge/core` (using the `@` notation) for further assistance or you can try [rerendering locally](https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-with-conda-smithy-locally).


The following suggestions might help debug any issues:
* Is the `recipe/{meta.yaml,recipe.yaml}` file valid?
* If there is a `recipe/conda-build-config.yaml` file in the feedstock make sure that it is compatible with the current [global pinnnings](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml).
* Is the fork used for this PR on an organization or user GitHub account? Automated rerendering via the webservices admin bot only works for user GitHub accounts.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/14862725282](https://github.com/conda-forge/conda-forge-webservices/actions/runs/14862725282). Examine the logs at this URL for more detail.</sub>

danielhollas commented on recipe/meta.yaml:
```diff
@@ -1,7 +1,10 @@
 # Set Version and Build-Number
-{% set version = "2.6.3" %}
+{% set version = "2.7.0rc0" %} # in source specified as 2.7.0pre0
```
```suggestion
{% set version = "2.7.0rc0" %}
```
Maybe this could fix the issue

agoscinski replied:
The last time it failed this was not there, but I can give it i a try. I wonder if the reason could that in the source code we use the identifier "2.7.0pre0" while in pypi it was translated to "2.7.0rc0".

agoscinski commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you but ran into some issues. Please check the [output logs](https://github.com/conda-forge/conda-forge-webservices/actions/runs/14864308698) of the GitHub Actions workflow below for more details. You can also ping `conda-forge/core` (using the `@` notation) for further assistance or you can try [rerendering locally](https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-with-conda-smithy-locally).


The following suggestions might help debug any issues:
* Is the `recipe/{meta.yaml,recipe.yaml}` file valid?
* If there is a `recipe/conda-build-config.yaml` file in the feedstock make sure that it is compatible with the current [global pinnnings](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml).
* Is the fork used for this PR on an organization or user GitHub account? Automated rerendering via the webservices admin bot only works for user GitHub accounts.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/14864308698](https://github.com/conda-forge/conda-forge-webservices/actions/runs/14864308698). Examine the logs at this URL for more detail.</sub>

danielhollas commented:
>  I wonder if the reason could that in the source code we use the identifier "2.7.0pre0" while in pypi it was translated to "2.7.0rc0".

I don't think that's the issue, but the `pip check` fails with

> aiormq 6.8.1 has requirement pamqp==3.3.0, but you have pamqp 3.2.1.

@conda-forge-admin, please  rerender

danielhollas commented on recipe/meta.yaml:
```diff
@@ -44,41 +47,43 @@ outputs:
+        - typing-extensions ~=4.0
         - upf_to_json ~=0.9.2
         - wrapt ~=1.11
+        - chardet ~=5.2.0
```
This is only needed on windows, I think this should work

```suggestion
        - chardet ~=5.2.0 [win]
```

agoscinski replied:
the linter complained because we use noarch, so I did not wanted to add it halfbaked

danielhollas replied:
Ah, okay, fair enough

agoscinski commented:
The issue I could identify was that the aiida meta package was now recognized as output (it was indexed) but we did not have any output specification. So I added something for the meta package, even though we should probably rather remove it.

@conda-forge-admin, please rerender

danielhollas commented:
Render succeeded!

Looks like the `pip check` is failing because `aiormq-feedstock` pins an old `pamqp` version.

pins https://github.com/conda-forge/aiormq-feedstock/blob/eddd7d2d364a1c073e8edf92fd6f6ab93eaa459c/recipe/meta.yaml#L24

The recipe needs to be updated there. The feedstock uses an automerge bot, so the python dependencies are not properly updated there. :cry:

agoscinski commented:
I put an ignore into the pip check for `pamqp` error as we cannot fix them

danielhollas commented:
> I put an ignore into the pip check for pamqp error as we cannot fix them

Well we should open a PR and ping the maintainers. :-) But for a rc0 release we can skip the check, sure.

danielhollas commented on recipe/meta.yaml:
```diff
@@ -98,15 +103,16 @@ outputs:
         - verdi --version
-        - pip check
+        # We cannot fix the fixed dependency in aiormq see https://github.com/conda-forge/aiormq-feedstock/issues/59 so we ignore it
+        - "! pip check | grep -v pamqp"
```
How about
```suggestion
        - pip check | test -z "$(grep -v pamqp)"
```

agoscinski replied:
fair try, but luckily CI passes^^

danielhollas commented:
Yes, let's squash.

Can you explain the metapackage a bit more? Why are the changes here suddenly needed? What is the purpose of the metapackage? Was it working up until now?

agoscinski commented:
> Can you explain the metapackage a bit more? Why are the changes here suddenly needed? What is the purpose of the metapackage? Was it working up until now?

I don't know mostly why. I just saw what caused the bug in the backtrace and by that could fix it but I don't know why the renderer suddenly picks up the meta package as output but does not have specs. But maybe the renderer logic has just changed. The linter complained the first time about things that were already present in old releases (e.g. the python min_version was not used), compare with last release ....  The metapackage contains aiida-core and the services.
```bash
$ conda search conda-forge::aiida=2.6.3 --info
Loading channels: done
aiida 2.6.3 pyhff2d567_0
------------------------
file name   : aiida-2.6.3-pyhff2d567_0.conda
name        : aiida
version     : 2.6.3
build       : pyhff2d567_0
build number: 0
size        : 7 KB
license     : MIT
subdir      : noarch
url         : https://conda.anaconda.org/conda-forge/noarch/aiida-2.6.3-pyhff2d567_0.conda
md5         : fdf44cdf47f7254cae2d0086f503a255
timestamp   : 2024-11-06 16:24:27 UTC
dependencies: 
  - aiida-core 2.6.3 pyhff2d567_0
  - aiida-core.services >=2.6.3,<3.0a0
  - python
```
I changed the pinning now to agree with this. But it is strange that this is required. I put it as TODO into issue https://github.com/conda-forge/aiida-core-feedstock/issues/107 with the other things of this PR to not forget it for the official release.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found some lint.

Here's what I've got...


For **recipe/meta.yaml**:

 * ❌ The feedstock has no `.ci_support` files and thus will not build any packages.

For **recipe/meta.yaml**:

 * ℹ️ It looks like the 'aiida' output doesn't have any tests.
 * ℹ️ `noarch: python` recipes should usually follow the syntax in our [documentation](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python) for specifying the Python version.
   - For the `test.requires` section of `<output 0` output, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 1` output, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 1` output, you should usually use the pin `python >={{ python_min }}` for the `python` entry.
   - For the `test.requires` section of `<output 1` output, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 2` output, you should usually use the pin `python >={{ python_min }}` for the `python` entry.
   - For the `test.requires` section of `<output 2` output, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 3` output, you should usually use the pin `python >={{ python_min }}` for the `python` entry.
   - For the `test.requires` section of `<output 3` output, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 4` output, you should usually use the pin `python >={{ python_min }}` for the `python` entry.
   - For the `test.requires` section of `<output 4` output, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 5` output, you should usually use the pin `python >={{ python_min }}` for the `python` entry.
   - For the `test.requires` section of `<output 5` output, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - If the package requires a newer Python version than the currently supported minimum version on `conda-forge`, you can override the `python_min` variable by adding a Jinja2 `set` statement at the top of your recipe (or using an equivalent `context` variable for v1 recipes).


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/14870158849](https://github.com/conda-forge/conda-forge-webservices/actions/runs/14870158849). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * ℹ️ `noarch: python` recipes should usually follow the syntax in our [documentation](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python) for specifying the Python version.
   - For the `test.requires` section of `<output 0` output, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - For the `host` section of `<output 1` output, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 1` output, you should usually use the pin `python >={{ python_min }}` for the `python` entry.
   - For the `test.requires` section of `<output 1` output, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 2` output, you should usually use the pin `python >={{ python_min }}` for the `python` entry.
   - For the `test.requires` section of `<output 2` output, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 3` output, you should usually use the pin `python >={{ python_min }}` for the `python` entry.
   - For the `test.requires` section of `<output 3` output, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 4` output, you should usually use the pin `python >={{ python_min }}` for the `python` entry.
   - For the `test.requires` section of `<output 4` output, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - For the `run` section of `<output 5` output, you should usually use the pin `python >={{ python_min }}` for the `python` entry.
   - For the `test.requires` section of `<output 5` output, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - If the package requires a newer Python version than the currently supported minimum version on `conda-forge`, you can override the `python_min` variable by adding a Jinja2 `set` statement at the top of your recipe (or using an equivalent `context` variable for v1 recipes).


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/14870254837](https://github.com/conda-forge/conda-forge-webservices/actions/runs/14870254837). Examine the logs at this URL for more detail.</sub>

danielhollas commented on recipe/meta.yaml:
```diff
@@ -98,15 +92,16 @@ outputs:
       run:
         - postgresql >=9.6
-        - rabbitmq-server >=3.7,<3.8.15
+        - rabbitmq-server >=3.11
```
Was this change needed? I think we need to be careful here. If we want to bump rabbitmq, we should try to somehow automatically configure it properly (i.e. override the timeout). Perhaps this should be done as a separate PR?

danielhollas replied:
I've reverted this change for now to unblock this PR since I think it requires more discussion. I've created #112 to discuss.

danielhollas reviewed (changes requested):
I think to get rid of the lint hints, you need to add `- python {{ python_min }}` to all the `test.requires` sections, see also PR #105

https://github.com/conda-forge/aiida-core-feedstock/pull/105/files#diff-f3725a55bf339595bf865fec73bda8ac99f283b0810c205442021f29c06eea9aR87

danielhollas commented:
> But this is already done in this PR or am I missing something?

You are :sweat_smile: You need to add explicit python requirement to the `test->requires` section, e.g. on line 80)

danielhollas commented:
@conda-forge-admin, please rerender

agoscinski commented:
> You are 😅 You need to add explicit python requirement to the test->requires section, e.g. on line 80)

Thanks! Now I see it. Added and squashed to one commit

@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/14881408345](https://github.com/conda-forge/conda-forge-webservices/actions/runs/14881408345). Examine the logs at this URL for more detail.</sub>

danielhollas reviewed (changes requested):
Thanks, the changes got rid of some lints, but there are still some remaining, see: https://github.com/conda-forge/aiida-core-feedstock/pull/106#issuecomment-2856119701

In particular we apparently need to add `python >={{ python_min }}` to all the run sections. Unfortunately I can't push to your branch, but I've pushed a commit to my fork that you can cherry-pick.

https://github.com/danielhollas/aiida-core-feedstock/tree/release/2.7.0rc0

I think we will not be able to get rid of all the lints. In particular, the `aiidaservices` target probably should not be `noarch: python` since neither postgresql nor rabbitmq are python packages. But let's leave that for another day. :sweat_smile:

agoscinski commented:
> Thanks, the changes got rid of some lints, but there are still some remaining, see: https://github.com/conda-forge/aiida-core-feedstock/pull/106#issuecomment-2856119701

I cherry-picked

> I think we will not be able to get rid of all the lints. In particular, the aiidaservices target probably should not be noarch: python since neither postgresql nor rabbitmq are python packages. But let's leave that for another day. 😅

Also we need to think about how to include the OS specific packages. I put these points into the issue #107

danielhollas reviewed (approved):
Looks like the remaining lints are all related to the aiidaservices output.
Thanks! Let's go! :rocket:

danielhollas commented:
My pamqp PT was merged so might be able to re-enable pip check here.

agoscinski commented:
@conda-forge-admin, please rerender

danielhollas reviewed (approved):
Let's go! :rocket:
